contrib/gitannex/git-annex-remote-rclone-goyle/main.go

```diff
@@ -0,0 +1,93 @@
+// # What even is this?
```

Comment:
Do you think this should be an rclone command so it is always built in to rclone? So something like `rclone git annexe` or something (don't really know what should be there!).

If not then this could live in its own repo quite easily? I'd be happy to host that repo under the `rclone` org.

Reply:
I thought this should be a subcommand originally, but it turns out that git-annex wants to exec `git-annex-remote-${type}` with no arguments, so there goes that idea. It seems a little silly to add a subcommand to rclone that requires a separate wrapper script/command to be useful.

> If not then this could live in its own repo quite easily? I'd be happy to host that repo under the rclone org.

Thanks for the offer to create a new repo! Unfortunately, my employer requires me to jump through more hoops if I'm creating a new project than if I'm patching an existing project. (I think even more hoops if I'm not releasing it under their GitHub org.) I'm going to research what the options are a little more and see how hard it might be to release in the rclone org.

So ultimately, I tried to thread the needle by putting this in contrib/,  satisfying both technical and bureaucratic constraints. Another advantage of this approach is that the project inherit's rclone's CI configuration and other infrastructure. I suppose the  disadvantage is that I have to bug you folks when I want to merge a PR, but it's nice to have another set of eyes :)

Reply:
> I thought this should be a subcommand originally, but it turns out that git-annex wants to exec `git-annex-remote-${type}` with no arguments, so there goes that idea. It seems a little silly to add a subcommand to rclone that requires a separate wrapper script/command to be useful.

Sorry, I forgot to mention the other part of the plan which would be to create a hardlink from `git-annex-remote-xxxx` to rclone then get rclone to start the correct subcommand. We already do this for [mount helpers](https://rclone.org/commands/rclone_mount/#rclone-as-unix-mount-helper) so this is easy to do.

> > If not then this could live in its own repo quite easily? I'd be happy to host that repo under the rclone org.
> 
> Thanks for the offer to create a new repo! Unfortunately, my employer requires me to jump through more hoops if I'm creating a new project than if I'm patching an existing project. (I think even more hoops if I'm not releasing it under their GitHub org.) I'm going to research what the options are a little more and see how hard it might be to release in the rclone org.

I could create the project under the rclone org then invite you to collaborate - perhaps that would satisfy your employers?

> So ultimately, I tried to thread the needle by putting this in contrib/, satisfying both technical and bureaucratic constraints. Another advantage of this approach is that the project inherit's rclone's CI configuration and other infrastructure. 

The CI is a lot of work to set up, I agree! If it is part of rclone it could also participate in the integration tests.

> I suppose the disadvantage is that I have to bug you folks when I want to merge a PR, but it's nice to have another set of eyes :)

:-)

Reply:
> Sorry, I forgot to mention the other part of the plan which would be to create a hardlink from `git-annex-remote-xxxx` to rclone then get rclone to start the correct subcommand. We already do this for [mount helpers](https://rclone.org/commands/rclone_mount/#rclone-as-unix-mount-helper) so this is easy to do.

Oh, cool! I didn't realize this was an option!

I think making this a subcommand is the best option, then. This also means I don't need to jump through approval hoops.

Thanks for the pointer to the mount helpers. IIUC, it looks like fs/mount_helpers.go transforms args in its `init()` *before*  rclone's `Main()` function runs. I can probably pull the same trick in the new subcommand's `init()`.

Reply:
> I think making this a subcommand is the best option, then. This also means I don't need to jump through approval hoops.

Great

> Thanks for the pointer to the mount helpers. IIUC, it looks like fs/mount_helpers.go transforms args in its init() before rclone's Main() function runs. I can probably pull the same trick in the new subcommand's init().

I can't remember exactly how that works, but I'm sure you'll figure it out!